### PR TITLE
Reenable working clfswm

### PR DESCRIPTION
**Pull request by michael-p-m-white**
###### Motivation for this change
I'm interested in trying out Common Lisp window managers, specifically CLFSWM. It appears that this window manager was never moved over to using quicklisp, as StumpWM (another Common Lisp window manager) was.

###### Things done
Added CLFSWM through quicklisp.
Added the 'clfswm' nixos module back in.
Updated the 'clfswm' nixos module to use the quicklisp CLFSWM, and not the normal (broken) CLFSWM package.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/lisp-modules/quicklisp-to-nix-output/clfswm.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+args @ { fetchurl, ... }:
+rec {
+  baseName = ''clfswm'';
+  version = ''20161204-git'';
+
+  description = ''CLFSWM: Fullscreen Window Manager'';
```
Comment (SuperSandro2000) [inline pkgs/development/lisp-modules/quicklisp-to-nix-output/clfswm.nix:6]:
```suggestion
  baseName = "clfswm";
  version = "20161204-git";

  description = "CLFSWM: Fullscreen Window Manager";
```

File: pkgs/development/lisp-modules/quicklisp-to-nix-output/clfswm.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  deps = [ args."clx" ];
+
+  src = fetchurl {
+    url = ''http://beta.quicklisp.org/archive/clfswm/2016-12-04/clfswm-20161204-git.tgz'';
+    sha256 = ''1jgz127721dgcv3qm1knc335gy04vzh9gl0hshp256rxi82cpp73'';
```
Comment (SuperSandro2000) [inline pkgs/development/lisp-modules/quicklisp-to-nix-output/clfswm.nix:12]:
```suggestion
    url = "http://beta.quicklisp.org/archive/clfswm/2016-12-04/clfswm-20161204-git.tgz";
    sha256 = "1jgz127721dgcv3qm1knc335gy04vzh9gl0hshp256rxi82cpp73";
```

Can we reuse version here?

**Review by SuperSandro2000 — COMMENTED** [review]
'' is now used for multiline strings in nix and " for single line.

**Comment by michael-p-m-white** [comment]
@SuperSandro2000 Regarding all your comments clfswm.nix: all .nix files in the quicklisp-to-nix-output directory are automatically generated by quicklisp-to-nix (see README.txt in lisp-modules, as well as the "quicklisp-to-nix" derivation in lisp-packages.nix in the same directory). quicklisp-to-nix uses a templating library to produce these .nix files, which uses double single-quoted strings. I'm not familiar enough with this implementation to know if there's a fundamental reason for this, or if the original author just happened to think this was a better idea, but I can say the template file has used these double single-quoted strings since this was originally implemented in March 2017.

If this is a problem, we should update quicklisp-to-nix and regenerate all packages in quicklisp-to-nix-output, which seems like it should be its own issue. Perhaps the original author (@7c6f434c) should weigh in?

**Comment by SuperSandro2000** [comment]
Didn't know that. You are right that it should be changed in the generator.

**Comment by 7c6f434c** [comment]
I am not even the original author of the generating scripts…

**Comment by michael-p-m-white** [comment]
@7c6f434c 

> I am not even the original author of the generating scripts…

Sorry; based on `git log -- /path/to/nix-package.emb`, it looked like you added this in commit 1300795. At any rate, you have the most commits against files in the quicklisp-to-nix directory, so I figured you would be the best person to weigh in on whether or not there were any blocking issues.

Cheers for the merge!

**Comment by 7c6f434c** [comment]
>@7c6f434c 
>
>> I am not even the original author of the generating scripts…
>
>Sorry; based on `git log -- /path/to/nix-package.emb`, it looked like you added this in commit 1300795. At any rate, you have the most commits against files in the quicklisp-to-nix directory, so I figured you would be the best person to weigh in on whether or not there were any blocking issues.

Ah, it does reuse my old template from the earlier generator

I am more or less sure that using " or '' based on presence of " in the upstream descriptions is not worth it, and then it is more escaping vs more abuse of '' in the code that is not supposed to be read, so I am not convinced " is even better here.

**Pull request merged**